Tests/Marketplace/search_and_install_packs.py

```diff
@@ -413,381 +600,348 @@ def install_packs(client: demisto_client,
-            server_min_version = pack_metadata.get(Metadata.SERVER_MIN_VERSION, Metadata.SERVER_DEFAULT_MIN_VERSION)
-            hidden = pack_metadata.get(Metadata.HIDDEN, False)
-            # Check if the server version is greater than the minimum server version required for this pack or if the
-            # pack is hidden (deprecated):
-            if ('Master' in server_version or Version(server_version) >= Version(server_min_version)) and not hidden:
-                logging.debug(f"Appending pack id {pack_id}")
-                all_packs.append(get_pack_installation_request_data(pack_id, pack_version))
-            else:
-                reason = 'Is hidden' if hidden else f'min server version is {server_min_version}'
-                logging.debug(f'Pack: {pack_id} with version: {pack_version} will not be installed on {host}. '
-                              f'Pack {reason}.')
-    return install_packs(client, host, all_packs)
-
+    Update 'packs_to_install' (a pointer to a list that's reused and updated by the function on every iteration)
+    with 'pack_id' and its dependencies, if 'pack_id' is not deprecated.
+    The way deprecation status is determined depends on the 'production_bucket' flag.
 
-def upload_zipped_packs(client: demisto_client,
-                        host: str,
-                        pack_path: str):
-    """
-    Install packs from zip file.
+    If 'production_bucket' is True, deprecation status is determined by checking the 'pack_metadata.json' file
+    in the commit hash that was used for the last upload. If it's False, the deprecation status is checking the
+    'pack_metadata.json' file locally (with changes applied on the branch).
 
     Args:
         client (demisto_client): The configured client to use.
-        host (str): The server URL.
-        pack_path (str): path to pack zip.
+        pack_id (str): The id of the pack to be installed.
+        production_bucket (bool): Whether pack deprecation status  is determined using production bucket.
+        commit_hash (str): Commit hash to use for checking pack's deprecations status if GitLab's API is used.
+            If 'pack_api_data' is not provided, will be used for fetching 'pack_metadata.json' file from GitLab.
     """
-    header_params = {
-        'Content-Type': 'multipart/form-data'
-    }
-    auth_settings = ['api_key', 'csrf_token', 'x-xdr-auth-id']
-    file_path = str(Path(pack_path).resolve())
-    files = {'file': file_path}
-
-    logging.info(f'Making "POST" request to server {host} - to install all packs from file {pack_path}')
+    if is_pack_deprecated(
+        pack_id=pack_id, production_bucket=production_bucket, commit_hash=commit_hash
+    ):
+        logging.warning(
+            f"Pack '{pack_id}' is deprecated (hidden) and will not be installed."
+        )
+        return pack_id, None  # Don't install deprecated packs
```

Comment:
Line 800 is like this:
```suggestion
        return pack_id, []  # Don't install deprecated packs
```

Reply:
this is the desired behaviour, either None for Error, or a list, that could be empty.